rye/src/cli/publish.rs

```diff
@@ -216,18 +360,21 @@ fn maybe_encrypt(secret: &Secret<String>, yes: bool) -> Result<Secret<Vec<u8>>,
     Ok(Secret::new(token.to_vec()))
 }
 
-fn maybe_decrypt(secret: &Secret<String>, yes: bool) -> Result<Secret<String>, Error> {
-    let phrase = if !yes {
-        dialoguer::Password::with_theme(tui_theme())
-            .with_prompt("Decrypt with passphrase (optional)")
-            .allow_empty_password(true)
-            .report(false)
-            .interact()
-            .map(Secret::new)?
+fn prompt_encrypt_passphrase() -> Result<Option<Secret<String>>, Error> {
```

Comment:
This is one comment but can also be made on a few other similar lines.

Note that prompt's return `Result<Option<_>>` when prompting. This function, for example, can allow empty inputs, but invoking the prompt from the code should either provide a passphrase or not.